lib/polyfill/media_capabilities.js

```diff
@@ -217,6 +234,51 @@ shaka.polyfill.MediaCapabilities = class {
     return res;
   }
 
+  /**
+   * Checks if the given media parameters of the video or audio streams are
+   * supported by the platform.
+   * @param {{
+   *   contentType: string,
+   *   width: (number|undefined),
+   *   height: (number|undefined),
+   *   frameRate: (number|undefined),
+   *   transferFunction: (string|undefined)
+   * }} options canDisplayType() options.
+   *     contentType: A valid MIME type and (optionally) a codecs parameter.
+   *     width: Describes the stream horizontal resolution in pixels.
+   *     height: Describes the stream vertical resolution in pixels.
+   *     frameRate: Describes the frame rate of the stream.
+   *     transferFunction: Describes the video transfer function supported by
+   *         the rendering capabilities of the user agent.
+   * @return {boolean} `true` when the stream can be displayed on a Cast device.
+   * @private
+   */
+  static canDisplayType_({
+    contentType,
+    width = undefined,
+    height = undefined,
+    frameRate = undefined,
+    transferFunction = undefined,
+  }) {
+    if (!cast.__platform__) {
```

Comment:
This is meant to be called only if the platform is cast, so I think instead of returning true here, it should throw an error.

If you remove this check, it would throw on non-cast devices naturally.

I would also suggest you rename this method to include the word "cast" so that the precondition of running on a cast platform were clearer.

Reply:
Makes sense, I've added a new error and renamed the function here to mention `Cast`. I also made the cast namespace check `cast`, `cast.__platform__`, and `cast.__platform__.canDisplayType`.